iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -137,90 +89,122 @@ static LogicalResult setContractConfig(FuncOp entryPoint, linalg::LinalgOp op) {
   invocationLevelTs.append(
       {tileX / workgroupSize[1], tileY / workgroupSize[0]});
   tileSizes.push_back(invocationLevelTs);  // Thread level.
-  setConfig(tileSizes, op);
-  return setTranslationInfo(entryPoint, passPipeline, workgroupSize);
+  return setOpConfigAndEntryPointFnTranslation(
+      entryPoint, op, tileSizes, /*nativeVectorSize=*/ArrayRef<int64_t>{},
+      IREE::HAL::DispatchLoweringPassPipeline::LLVMGPUVectorize, workgroupSize);
 }
 
 // Basic default properties for linalg ops that haven't been tuned.
-static LogicalResult setRootDefaultConfig(FuncOp entryPoint,
-                                          linalg::LinalgOp op) {
+static LogicalResult setRootDefaultConfig(FuncOp entryPoint, Operation *op) {
   IREE::HAL::DispatchLoweringPassPipeline passPipeline =
       IREE::HAL::DispatchLoweringPassPipeline::LLVMGPUDistribute;
-  std::array<int64_t, 3> workgroupSize = {1, 1, 1};
   TileSizesListType tileSizes;
-  size_t numLoops = getNumOuterParallelLoops(op);
-  if (numLoops == 0) {
-    return setTranslationInfo(entryPoint, passPipeline, workgroupSize);
+  SmallVector<unsigned> partitionedLoops = getPartitionedLoops(op);
+  if (partitionedLoops.empty()) {
+    tileSizes.push_back({});
+    return setOpConfigAndEntryPointFnTranslation(
+        entryPoint, op, tileSizes, /*nativeVectorSize=*/ArrayRef<int64_t>{},
+        passPipeline, {1, 1, 1});
   }
 
-  workgroupSize = {cudaWarpSize, 1, 1};
-  int64_t lowerTs = 4 * cudaWarpSize;
-  SmallVector<int64_t, 4> ts;
-  ts.resize(numLoops, 1);
-  ts.back() = lowerTs;
-  tileSizes.push_back(ts);
-  tileSizes.push_back({});  // Subgroup level.
-  ts.back() = lowerTs / cudaWarpSize;
-  tileSizes.push_back(ts);  // Thread level.
-  setConfig(tileSizes, op);
-  return setTranslationInfo(entryPoint, passPipeline, workgroupSize);
+  size_t numLoops = partitionedLoops.back() + 1;
+
+  std::array<int64_t, 3> workgroupSize = {cudaWarpSize, 1, 1};
+  static constexpr unsigned vectorSize = 4;
+  SmallVector<int64_t, 4> workgroupTileSizes(numLoops, 1),
+      threadTileSizes(numLoops, 1);
+  // Set all non-parallel loops to zero tile size.
+  llvm::DenseSet<unsigned> partitionedLoopsSet(partitionedLoops.begin(),
+                                               partitionedLoops.end());
+  for (auto depth : llvm::seq<int64_t>(0, numLoops)) {
+    if (!partitionedLoopsSet.count(depth)) {
+      workgroupTileSizes[depth] = 0;
```

Comment:
Question: Seems like tile sizes used to default to 1, is there any particular reason that we are setting it the default to 0 now?